### PR TITLE
fix: parse commit hash of GPG signed commit

### PR DESCRIPTION
**Pull request by sublimino**
This change will support parsing the commit hash from GPG-signed commits. It does so by disabling signing output from the `git` CLI using `git -c log.showSignature=false` prior to the git subcommand.

@controlplaneio security policy mandates GPG signed commits, and we'd like to use this tool.

Closes https://github.com/goreleaser/goreleaser/issues/952

### CODE REVIEW

**Comment by caarlos0** [comment]
wouldn't this work too? https://github.com/goreleaser/goreleaser/compare/master...quiet?expand=1

File: internal/testlib/git.go [hunk 1]
```diff
@@ -48,6 +48,7 @@ func fakeGit(args ...string) (string, error) {
 		"-c", "user.name='GoReleaser'",
 		"-c", "user.email='test@goreleaser.github.com'",
 		"-c", "commit.gpgSign=false",
+		"-c", "log.showSignature=false",
```
Comment (caarlos0) [inline internal/testlib/git.go]: this is not needed I believe...

**Comment by sublimino** [comment]
> wouldn't this work too? [goreleaser/goreleaser/compare/master...quiet?expand=1](https://github.com/goreleaser/goreleaser/compare/master...quiet?expand=1)

Not on latest Debian with git version 2.20.1. The flag in this PR is needed. 

`-q` can be added to suppress further lines (although it seems they're discarded anyway).

```
$ git show --format='%H' HEAD
gpg: Signature made Fri 11 Jan 2019 16:50:03 CET
gpg:                using RSA key 7DCF335A3BB2EC440C516D4C7B9EA6DD705D5F09
gpg: Good signature from "Andrew Martin <sublimino@gmail.com>" [ultimate]
e0754ec9a9cfc8cd9fe2de6b918235b44da916d1

diff --git a/x b/y
similarity index 100%
rename from x
rename to y
```

```
$ git show --format='%H' HEAD -q
gpg: Signature made Fri 11 Jan 2019 16:50:03 CET
gpg:                using RSA key 7DCF335A3BB2EC440C516D4C7B9EA6DD705D5F09
gpg: Good signature from "Andrew Martin <sublimino@gmail.com>" [ultimate]
e0754ec9a9cfc8cd9fe2de6b918235b44da916d1
```

```
$ git -c log.showSignature=false show --format='%H' HEAD
e0754ec9a9cfc8cd9fe2de6b918235b44da916d1

diff --git a/x b/y
similarity index 100%
rename from x
rename to y
```

```
$ git -c log.showSignature=false show --format='%H' HEAD -q
e0754ec9a9cfc8cd9fe2de6b918235b44da916d1
```

**Comment by caarlos0** [comment]
oh, ok, thanks for testing it... I couldn't reproduce this locally, even with signed commits... anyway, will merge this in.

thanks for the PR :)

**Comment by sublimino** [comment]
Ah yes, that's because my `~/.gitconfig` is set to show signatures. Thanks for the merge :pray: 

```toml
[log]              
  showSignature = true   
```

**Comment by caarlos0** [comment]
hmmm, that makes sense... anyway, it safer to prevent that =D

**Pull request merged**